cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -3970,6 +3970,9 @@ macro(build_google_cloud_cpp_storage)
   if(ABSL_VENDORED)
     list(APPEND GOOGLE_CLOUD_CPP_PREFIX_PATH_LIST ${ABSL_PREFIX})
   endif()
+  if(TARGET zlib_ep)
```

Comment:
Could you add `set(ZLIB_VENDORED TRUE)` to `build_zlib()` and use it instead of `TARGET zlib_ep` like others?